e2e_test/udf/sql_udf.slt

```diff
@@ -0,0 +1,105 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+# Create an anonymous function with double dollar as clause
+statement ok
+create function add(INT, INT) returns int as $$select $1 + $2$$ language sql;
+
+# Create an anonymous function with single quote as clause
+statement ok
+create function sub(INT, INT) returns int as 'select $1 - $2' language sql;
+
+# Currently we can only support constant calling convention
+statement ok
+create function add_sub_binding() returns int as 'select add(1, 1) + sub(2, 2)' language sql;
+
+# Call the defined sql udf
+query I
+select add(1, -1);
+----
+0
+
+query I
+select sub(1, 1);
+----
+0
+
+query I
+select add_sub_binding();
+----
+2
+
+query III
+select add(1, -1), sub(1, 1), add_sub_binding();
+----
+0 0 2
+
+# Create a mock table
+statement ok
+create table t1 (c1 INT, c2 INT);
+
+# Insert some data into the mock table
+statement ok
+insert into t1 values (1, 1), (2, 2), (3, 3), (4, 4), (5, 5);
+
+query III
+select sub(c1, c2), c1, c2, add(c1, c2) from t1 order by c1 asc;
+----
+0 1 1 2
+0 2 2 4
+0 3 3 6
+0 4 4 8
+0 5 5 10
+
+# Invalid function body syntax
+statement error
+create function add_error(INT, INT) returns int as $$select $1 + $2 +$$ language sql;
+
+# Multiple type interleaving sql udf
+statement ok
+create function add_sub(INT, FLOAT, INT) returns FLOAT as $$select -$1 + $2 - $3$$ language sql;
+
+# Note: need EXPLICIT type cast in order to call the multiple types interleaving sql udf
```

Comment:
Please add a case for the error if type cast absent

Reply:
In fact, this is due to a flaw in our implementation #9998. The error is not expected.

Reply:
That's OK and can leave a TODO. I want to have an error test, which can also kind of act as a doc of our current behavior.